e2e-tests/webdriver-helpers.ts

```diff
@@ -95,141 +101,185 @@ export class WebDriverHelper {
-    let tries = WAIT_FOR_SUBMISSION_TRIES;
-    do {
-      // Wait for the first LOI submission.
-      try {
+    try {
+      assertWebDriverInitialized(this.driver);
+      let lastError: Error | null = null;
+      let tries = WAIT_FOR_SUBMISSION_TRIES;
+      do {
+        // Wait for the first LOI submission.
         try {
-          await this.waitUntilPresent(By.css('.loi-icon'), SHORT_TIMEOUT);
-        } catch (e) {
-          const expandButtonSelector = By.css(
-            '.job-list-item-container button'
-          );
-          await this.waitUntilPresent(expandButtonSelector, LONG_TIMEOUT);
-          const expandSubmissionsButton = await this.driver.findElement(
-            expandButtonSelector
+          try {
+            await this.waitUntilPresent(By.css('.loi-icon'), SHORT_TIMEOUT);
+          } catch (e) {
+            const expandButtonSelector = By.css(
+              '.job-list-item-container button'
+            );
+            await this.waitUntilPresent(expandButtonSelector, LONG_TIMEOUT);
+            const expandSubmissionsButton = await this.driver.findElement(
+              expandButtonSelector
+            );
+            await expandSubmissionsButton.click();
+            await this.waitUntilPresent(By.css('.loi-icon'), LONG_TIMEOUT);
+          }
+          // Wait for expected number of submissions.
+          const loiIcon = await this.driver.findElement(By.css('.loi-icon'));
+          await loiIcon.click();
+          await this.waitUntilPresent(By.css('.submission-item'));
+          const submissionItems = await this.driver.findElements(
+            By.css('.submission-item')
           );
-          await expandSubmissionsButton.click();
-          await this.waitUntilPresent(By.css('.loi-icon'), LONG_TIMEOUT);
-        }
-        // Wait for expected number of submissions.
-        const loiIcon = await this.driver.findElement(By.css('.loi-icon'));
-        await loiIcon.click();
-        await this.waitUntilPresent(By.css('.submission-item'));
-        const submissionItems = await this.driver.findElements(
-          By.css('.submission-item')
-        );
-        if (submissionItems.length < EXPECTED_SUBMISSION_COUNT) {
-          await this.delay(LONG_TIMEOUT);
-          throw new WebDriverHelperException('Not enough survey submissions');
+          if (submissionItems.length < EXPECTED_SUBMISSION_COUNT) {
+            await this.delay(LONG_TIMEOUT);
+            throw new WebDriverHelperException('Not enough survey submissions');
+          }
+          return;
+        } catch (e) {
+          lastError = e as Error;
         }
-        return;
-      } catch (e) {
-        lastError = e as Error;
-      }
-    } while (tries-- > 0);
-    fail(`No survey submissions appeared: ${lastError}`);
+      } while (tries-- > 0);
+      fail(`No survey submissions appeared: ${lastError}`);
+    } catch (e) {
+      console.error(e);
+      console.log((e as Error).stack);
```

Comment:
I believe .error() shows the stack already - we shouldn't need both .log and .error?

Reply:
You are totally right! Done.